core/embed/rust/src/ui/component/model_t1/dialog.rs

```diff
@@ -0,0 +1,116 @@
+use super::{
+    button::{Button, ButtonMsg::Clicked},
+    theme,
+};
+use crate::{
+    micropython::buffer::Buffer,
+    ui::{
+        component::{Child, Component, Event, EventCtx},
+        display,
+        geometry::{Point, Rect},
+    },
+};
+
+pub enum DialogMsg<T> {
+    Content(T),
+    LeftClicked,
+    RightClicked,
+}
+
+pub struct Dialog<T, U> {
+    header: Option<Buffer>,
+    content: Child<T>,
+    left_btn: Option<Child<Button<U>>>,
+    right_btn: Option<Child<Button<U>>>,
+}
+
+impl<T: Component, U: AsRef<[u8]>> Dialog<T, U> {
+    pub fn new(
+        area: Rect,
+        content: impl FnOnce(Rect) -> T,
+        left: Option<impl FnOnce(Rect) -> Button<U>>,
+        right: Option<impl FnOnce(Rect) -> Button<U>>,
+        header: Option<Buffer>,
+    ) -> Self {
+        let button_h = 11;
+        let header_h = 13;
```

Comment:
Similarly to the comment over `Button::paint`, could we move this magic numbers somewhere?